src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -516,16 +534,13 @@ private void PruneCache()
                 }
             }
 
-            foreach (TrieNode trieNode in toRemove)
+            for (int index = 0; index < _nodesToRemoveOnPruning.Count; index++)
             {
-                if (trieNode.Keccak is null)
```

Comment:
why this check is no longer needed?

Reply:
@LukaszRozmej Is this answered/addressed?

Reply:
_nodesToRemoveOnPruning now contain keys only

Reply:
Reverted